torch/_inductor/codegen/wrapper.py

```diff
@@ -1514,6 +1521,23 @@ def generate_numel_expr(self, kernel_name: str, tree, suffix: Optional[str] = No
         # it suffices as a type hint for the purposes of producing the correct code for this type.
         return SymbolicCallArg(expr, tree.numel)
 
+    def generate_semaphores_allocation(self, count, device):
```

Comment:
Is there any particular reason that semaphore allocation needs to be different than workspace allocation?

Reply:
One needs to be zeroed and the other does not, so I'm trying to avoid extra kernel launches to zero things out.

Reply:
I think `generate_workspace_allocation` has an option for whether or not to zero it out.

https://github.com/pytorch/pytorch/blob/5fee1ee3f4aed574df30720ee796b4560e550893/torch/_inductor/codegen/wrapper.py#L1517

Reply:
Yes, it does, but if it used it:
1) There one be one zeroing out per call (rather than per graph)
2) I would be zeroing out unneeded stuff.  I might have 2 semaphores needing zeroing + hundreds of bytes of scratch space that *don't* need zeroing.

Reply:
I broke this stuff into its own PR #138165